academic-observatory-workflows/academic_observatory_workflows/crossref_metadata_telescope/telescope.py

```diff
@@ -0,0 +1,326 @@
+            "transform": {"memory": "16G", "cpu": "16"},
+            "upload_transformed": {"memory": "8G", "cpu": "8"},
+        }
+        if overrides:
+            for task, resources in overrides.items():
+                task_resources[task] = resources
+
+        self.download = V1ResourceRequirements(requests=task_resources["download"], limits=task_resources["download"])
+        self.upload_downloaded = V1ResourceRequirements(
+            requests=task_resources["upload_downloaded"], limits=task_resources["upload_downloaded"]
+        )
+        self.extract = V1ResourceRequirements(requests=task_resources["extract"], limits=task_resources["extract"])
+        self.transform = V1ResourceRequirements(
+            requests=task_resources["transform"], limits=task_resources["transform"]
+        )
+        self.upload_transformed = V1ResourceRequirements(
+            requests=task_resources["upload_transformed"], limits=task_resources["upload_transformed"]
+        )
+
+
+@dataclass
+class DagParams:
+    """Parameters for the Crossref Metadata Telescope
+
+    :param dag_id: the id of the DAG.
+    :param cloud_workspace: the cloud workspace settings.
+    :param bq_dataset_id: the BigQuery dataset id.
+    :param bq_table_name: the BigQuery table name.
+    :param api_dataset_id: the Dataset ID to use when storing releases.
+    :param schema_folder: the SQL schema path.
+    :param dataset_description: description for the BigQuery dataset.
+    :param table_description: description for the BigQuery table.
+    :param crossref_metadata_conn_id: the Crossref Metadata Airflow connection key.
+    :param observatory_api_conn_id: the Observatory API connection key.
+    :param max_processes: the number of processes used with ProcessPoolExecutor to transform files in parallel.
+    :param batch_size: the number of files to send to ProcessPoolExecutor at one time.
+    :param start_date: the start date of the DAG.
+    :param schedule: the schedule interval of the DAG.
+    :param catchup: whether to catchup the DAG or not.
+    :param queue: what Airflow queue this job runs on.
+    :param max_active_runs: the maximum number of DAG runs that can be run at once.
+    :param retries: the number of times to retry a task.
+    :param gke_image: The image location to pull from.
+    :param gke_namespace: The cluster namespace to use.
+    :param gke_startup_timeout_seconds: How long to wait for the container to start in seconds.
+    :param gke_conn_id: The name of the airlfow connection storing the gke cluster information.
+    :param docker_astro_uuid: The uuid of the astro user
+    :param gke_volume_name: The name of the persistent volume to create
+    :param gke_volume_path: Where to mount the persistent volume
+    :param gke_zone: The zone containing the gke cluster
+    :param gke_volume_size: The amount of storage to request for the persistent volume in GiB
+    :param gke_resource_overrides: Task resource overrides
```

Comment:
Could these be in their own data class and added as a single parameter to this one? Since they will be re-used across a number of different workflows, with the same settings.